onnx/defs/tensor/defs.cc

```diff
@@ -682,14 +687,22 @@ ONNX_OPERATOR_SET_SCHEMA(
                   ")");
             }
           } else { // no value available for 'split'
-            int num_outputs = static_cast<int>(ctx.getNumOutputs());
-            if (split_dim_value % num_outputs != 0) {
-              fail_shape_inference("The input is not evenly splittable");
-            }
-            int chunk_size = split_dim_value / num_outputs;
-            split.reserve(ctx.getNumOutputs());
-            for (int i = 0; i < static_cast<int>(ctx.getNumOutputs()); i++) {
-              split.push_back(chunk_size);
+            const auto num_outputs_attr = ctx.getAttribute("num_outputs");
```

Comment:
Suggest moving the above line out of the else-branch to before the if-statement, and adding a check in the then-branch that `num_outputs_attr == nullptr` (to catch the case where both are specified, which is also erroneous).

Reply:
I added a shape inference error in case both are specified. The PR has been moved to https://github.com/onnx/onnx/pull/4389.